contracts/Overtime/SportsAMMV2.sol

```diff
@@ -302,32 +338,53 @@ contract SportsAMMV2 is Initializable, ProxyOwned, ProxyPausable, ProxyReentranc
         uint _expectedPayout,
         address _differentRecipient,
         address _referrer,
-        address _collateral
-    ) external nonReentrant notPaused {
-        require(msg.sender == liveTradingProcessor, "only possible from live trading processor");
+        address _collateral,
+        bool _isETH
+    ) external payable nonReentrant notPaused {
+        require(msg.sender == liveTradingProcessor, "Only Live");
 
         if (_referrer != address(0)) {
             referrals.setReferrer(_referrer, _requester);
         }
 
-        require(_differentRecipient != address(0), "recipient has to be defined");
-
+        require(_differentRecipient != address(0), "undef Recip");
+        address useLPpool;
+        uint collateralPriceInUSD;
         if (_collateral != address(0)) {
-            _handleDifferentCollateral(_buyInAmount, _collateral, false, _requester);
+            (useLPpool, collateralPriceInUSD) = _handleDifferentCollateral(_buyInAmount, _collateral, msg.sender, _isETH);
         }
 
-        _trade(
-            _tradeData,
-            TradeDataInternal(
-                _buyInAmount,
-                _expectedPayout,
-                0, // no additional slippage allowed as the amount comes from the LiveTradingProcessor
-                _differentRecipient,
-                _collateral == address(0),
-                true,
-                _requester
-            )
-        );
+        if (useLPpool != address(0)) {
```

Comment:
feels like a lot of the code is similar to what happens in main trade method, so maybe extract it all to a private function